Mathlib/NumberTheory/MulChar/Lemmas.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.MulChar.Basic
+import Mathlib.RingTheory.Ideal.LocalRing
+import Mathlib.RingTheory.RootsOfUnity.Complex
+
+/-!
+# Further Results on multiplicative characters
+-/
+
+namespace MulChar
+
+/-- Two multiplicative characters on a monoid whose unit group is generated by `g`
+are equal if and only if they agree on `g`. -/
+lemma eq_iff {R R' : Type*} [CommMonoid R] [CommMonoidWithZero R'] {g : Rˣ}
+    (hg : ∀ x, x ∈ Subgroup.zpowers g) (χ₁ χ₂ : MulChar R R') :
+    χ₁ = χ₂ ↔ χ₁ g.val = χ₂ g.val := by
+  rw [← Equiv.apply_eq_iff_eq equivToUnitHom, MonoidHom.eq_iff_eq_on_generator hg,
+    ← coe_equivToUnitHom, ← coe_equivToUnitHom, Units.ext_iff]
+
+
+section Ring
+
+variable {R R' : Type*} [CommRing R] [CommRing R']
+
+/-- Define the conjugation (`star`) of a multiplicative character by conjugating pointwise. -/
+@[simps!]
+def starComp [StarRing R'] (χ : MulChar R R') : MulChar R R' :=
+  χ.ringHomComp (starRingEnd R')
+
+
+variable [Fintype R] [DecidableEq R]
+
+/-- The values of a multiplicative character on `R` are `n`th roots of unity, where `n = #Rˣ`. -/
+lemma val_mem_rootsOfUnity (a : Rˣ) {χ : MulChar R R'} :
```

Comment:
`val` is usually the name component that refers to `Subtype.val`.
I think in this case `apply` is what should be used.

Idem for the `val`s in lemmas below.

Reply:
OK, thanks for pointing this out!